sql/2022/jamstack/candidate_urls.sql

```diff
@@ -0,0 +1,88 @@
+-- step 1.1: get URLs with age headers
+with headers as (
+  SELECT 
+    url,
+    JSON_EXTRACT_ARRAY(payload, '$.response.headers') as headers_array
+  FROM `httparchive.requests.2022_07_01_mobile` 
+),

```

Comment:
This is available as a field in `httparchive.almanac.requests` which makes it significantly cheaper to query. The `almanac` dataset is an optimised version of the monthly datasets with certain keyfields (like this one) pulled out form the `payload` body to make them easier (and quicker and cheaper) to query.

Also we are using June not July for the almanac this year.

Plus we should look at both mobile and desktop.

```suggestion
with headers as (
  SELECT
    client,
    url,
    response_headers as headers_array
  FROM
    `httparchive.almanac.requests`
  WHERE
    date = '2022-06-01'
),
```

This is 1 TB compared to 12 TB.

And you will probably be able to merge this CTE into the next one now.